resources/digital_systems.md

```diff
@@ -0,0 +1,226 @@
+| There is a weight associated with each digit, dependant on the number's position in the sequence. | Digit Value is independent of a number's position in the sequence. Gray Code is one example of this. | _k-th_ smallest number is the complement of the _K-th_ largest number. Look up the 5211 code or 2421 code. | For every subsequent number, the binary representation only differs by one digit. Excess-3 and 8421 are codes of this type. | The family of binary codes used to represent alphanumeric data. The primordial example would be the ASCII encoding. | The family of binary encoding is meant to detect and or correct failed data transmission. Parity Checking is one example. |
+
+---
+
+## Multiplexer
+
+A multiplexer is a combinational circuit that selects one of many input lines
+and directs it to a single output line.
+The selection of a particular input line is controlled by a set of selection
+lines. Normally, there are $2^𝑛$ input
+lines and 𝑛 selection lines whose bit combinations determine which input is
+selected.
+
+An example of a 4 to 1 multiplexer is provided below.
+
+![4 to 1 Multiplexer.png](/resources/pictures/4%20to%201%20Multiplexer.png)
+
+---
+
+## Demuxes
+
+> Basic Multiplexer System. Permission allowed since attribution is fulfilled.
+> ![Telephony_multiplexer_system.gif](/resources/pictures/Telephony_multiplexer_system.gif)
+> Tony R. Kuphaldt, CC BY 1.0 <https://creativecommons.org/licenses/by/1.0>, via
+> Wikimedia Commons
+
+Demuxes perform the inverse operation of multiplexers: they direct a single
+input line to many output lines. The
+selection of a particular output line is controlled by a set of selection lines.
+Normally, there are $2^𝑛$ output lines
+and 𝑛 selection lines whose bit combinations determine which output is selected.
+
+---
+
+## Decoders
+
+Decoders can be imagined like a demux, where only select lines are present.
+Confusingly, these select lines are called
+input lines, so keep that in mind. When a partigular output gate is selected, a
+high signal (1) is continously
+transmitted along said line.
+
+Even more confusing, is that sometimes decoders have an enable input, which
+causes them to behave identically to a
+demux.
+
+The only real difference lies in how the gates are named.
+
+---
+
+## Finite State Machines
+
+A finite state machine is the next level up in digital design. State Machines
+Circuits will produce different outputs
+not just from the different combinations of their circuits, but also based on
+the output they were sending before, or
+just their outputs alone, with no input change at all!
+
+Finite State Machines can be grouped into two types: Mealy or Moore. A block
+diagram of each is presented below.
+
+![Mealy vs Moore Machines.png](/resources/pictures/Mealy%20vs%20Moore%20Machines.png)
```

Comment:
Same as comment on lines +160 to +163.

Reply:
The snippets were an oversight. I wanted to add my own tables myself. That has been done now.